cloudify_aws/codepipeline/resources/pipeline.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2018 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+    CodePipeline.Pipeline
+    ~~~~~~~~~~~~~~
+    AWS pipeline interface
+"""
+# Third party imports
+from botocore.exceptions import ClientError
+
+# Cloudify
+from cloudify_aws.common import decorators, utils
+from cloudify_aws.codepipeline import CodePipelineBase
+
+RESOURCE_TYPE = 'CodePipeline pipeline'
+RESOURCE_NAME = 'name'
+CREATED_STATUS = 'Created'
+
+
+class CodePipelinePipeline(CodePipelineBase):
+    """
+        AWS CodePipeline pipeline interface
+    """
+
+    def __init__(self, ctx_node, resource_id=None, client=None, logger=None):
+        CodePipelineBase.__init__(self, ctx_node, resource_id, client, logger)
+        self.type_name = RESOURCE_TYPE
+
+    @property
+    def properties(self):
+        """Gets the properties of an external resource"""
+        resources = None
+        try:
+            resources = self.client.get_pipeline_state(
+                name=self.resource_id)
+        except ClientError:
+            pass
+        return resources
+
+    @property
+    def status(self):
+        """Gets the status of an external resource"""
+        props = self.properties
+        if props:
+            if props.get('created'):
+                return CREATED_STATUS
+        return None
+
+    def create(self, params):
+        """
+            Create a new Pipeline .
+        """
+        return self.make_client_call('create_pipeline', params)
+
+    def delete(self, params=None):
+        """
+            Deletes an existing Pipeline.
+        """
+        self.logger.debug('Deleting {resource_type} with parameters:'
+                          ' {params}'.format(
+                                            resource_type=self.type_name,
+                                            params=params))
+        self.client.delete_pipeline(**params)
```

Comment:
Do we want to log result?

Reply:
I don't think its needed in the delete. If there is an error the user will see it.